src/sentry.ts

```diff
@@ -169,11 +170,14 @@ export default class SentryService {
       ]);
 
       try {
+        // set both filtered args and flags as tags for sentry
         const filtered_sentry_args = await this.filterNonSensitiveSentryMetadata(non_sensitive, args);
-        const filtered_sentry_flags = await this.filterNonSensitiveSentryMetadata(non_sensitive, flags);
+        const filtered_sentry_args_tags = this.flattenNestedJson(filtered_sentry_args, 'command_args');
```

Comment:
That's bad naming of the function on my part. The filter non sensitive returns a potentially nested json object, so this function is called to turn them into a flat k/v set of tags, eg
```
os_info {
  shell: /bin/zsh,
  pid: 0,
   ... 
   ... 
  }
```
Becomes a flat json when uploaded to sentry become separate tag entries:
* `os_info.shell: 'zsh'`
* `os_info.gid:  0`

Reply:
Ah gotcha. Still unsure if we should be setting them all as tags. Shouldn't we be more selective so we don't have to pick through hundreds of options in sentry? Also should we still set extra data for viewing on the error page in sentry so I don't have to comb through tags?

Reply:
I can revert the conversion of additional_data to tags, so it works how it is currently. Would it instead be better to add args and flags as tags? We can still use the function to parse the values, but it should lead to far fewer tags this way.

Reply:
+1 to still keeping the data as "extra data" for viewing in the Sentry body, it's more organized than the tags UI so it'd be good to have them in both places for viewing/filtering.

I also am not entirely sure if we want _every_ non sensitive flag/arg as a tag... I don't know if there's any sort of tag limit but it's possible that we'll get more useless info vs. additional insights.

A potential option would be to just set a `command.flags` and `command.args` tag and set that, so if I run

`architect deploy --account tyler-is-da-man --environment the-moon my_component`
we would get tags like:
`command.args: my_component`
`command.flags: --account tyler-is-da-man --environment the-moon`

(This may be a bad idea depending on how Sentry search works, we'd have to search "does command.flags contain --account" which may be harder than just if the tag exists)

Reply:
We mostly care about what commands are being used. I don't think we care about args since its user input. Flags we potentially care about - in terms of what key was used, but not the value.

Reply:
There appears to be no tag limit, so it could grow very large, up to N length where N is the number of unique command flag's.  eg. `command.flags.environment=myenvironment` and `command.flags.environment=architect` is stored as the same tag with a different value. args would populate similarly, with the key being `command.args.${arg_name} = value`.  I agree flags being used  would be most useful, and afterwards in decreasing usefullness flag values, args, and then the rest of the additional data. 

If we're not providing values for the flags, would it make sense to instead to set as true, for when its present and otherwise false? eg. `command.flags.account=true` `command.flags.force-apply=false` etc?

Reply:
Ah I clicked on the actual ticket and see the screenshot now with all the metadata as tags which is definitely overkill. e.g. a tag for what linked components somebody has isn't needed at all as it's just a filesystem path 😛. 

I do think if the primary bit of metadata we want to catalogue is which flags were used, and there's not really a tag limit to worry about, we should just add keys like `flag.environment: the-moon`, `flag.account: tyler-roolz` to the tags and not add _all_ of what is currently in the additional properties

Reply:
(IMO we may as well add the actual flag data instead of just 'true' if it was used, just in case it's useful in some way - it implies "true" and "false" would  be noisy to set)